docs/cartesian-product.adoc

```diff
@@ -300,37 +300,190 @@ To demonstrate with a table:
 
 For more information, please see the link:docs/range-sources[separate documentation about range sources].
 
+== Annotating your test method
+
+You can annotate your test method to supply arguments to all parameters simultaneously.
+
+=== `@CartesianTest.Factory`
+
+`@CartesianTest.Factory` can be used to name a static factory method that supplies your arguments.
+The `name` annotation parameter is mandatory.
+Just like with JUnit's `@MethodSource`, you can specify the factory method with its fully-qualified name (including the class), e.g. `com.example.Class#factory`.
+This method must return `CartesianMethodArgumentsProvider.Sets`.
+`Sets` is a helper class, specifically for creating sets for `@CartesianTest`.
+To create the test data, instantiate a `new Sets()` then use the `add()` method to register the values for the parameters.
```

Comment:
It should be made clearer that `add` needs to be called once per parameter with all arguments for that parameter.

Reply:
Updated it a bit.

Reply:
Me too. 😁